### PR TITLE
reintroduced _self (returns the current template name)

### LINKED ISSUE #1932: [Suggestion] Add a way to access the name of the current template
``_self`` accessing the current Twig_Template instance is deprecated (and removed in 2.0). This causes headaches in several places needing an easy way to reference the current template (and this is why we kept a special handling of it in from and import tags).

However, most places relying on ``_self`` today don't actually need the template instance. The template name would be enough as they pass it to functions resolving the template anyway. Would you accept a way to get the name of the current template rather than the instance (there is no security implication with just the name). This could even allow to remove ``_self`` entirely as the from and import tag would also accept it.

See https://github.com/symfony/symfony-demo/issues/242#issuecomment-153388234 and https://github.com/symfony/symfony/issues/16804 for instance

### PR DESCRIPTION
**Pull request by fabpot**
fixes #1932, see #1964 fro the 1.x counterpart.

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,9 +1,9 @@
-* 2.0.0 (201X-XX-XX)
+* 2.0.0 (2016-XX-XX)
 
  * sped up the json_encode filter
  * removed reserved macro names; all names can be used as macro
  * removed Twig_Template::getEnvironment()
- * removed _self variable (except for usage in from and import tags)
+ * changed _self variable to returns the current template name
```
Comment (Tobion) [inline CHANGELOG]: typo "return"

File: doc/templates.rst [hunk 2]
```diff
@@ -127,6 +127,7 @@ Global Variables
 
 The following variables are always available in templates:
 
+* ``_self``: reference the current template name;
```
Comment (Tobion) [inline doc/templates.rst]: "references" as below

File: CHANGELOG (hunk 1 above)
Comment (fabpot) [inline CHANGELOG]: fixed

File: doc/templates.rst (hunk 2 above)
Comment (fabpot) [inline doc/templates.rst]: fixed

**Pull request merged**